### PR TITLE
JSON to Shopify - Products

### LINKED ISSUE #119: Products import to Shopify (JSON)
1. Read a JSON file containing product data for Shopify 
2. Import into Shopify 

---
Parameters
- JSON file location
- Shopify creds

### PR DESCRIPTION
**Pull request by FrendsSarlinS**
Closes #119 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a template for importing products from a local JSON file to Shopify, including detailed instructions and error handling.
  - Added metadata support for the new "JSON to Shopify - Products" tool, specifying its version and system use.

### CODE REVIEW

**Review by jannejjj — APPROVED** [review]
Looks good and tested to work!

File: Templates/Files/Local Files/JSON to Shopify - Products/long-description.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+This template assumes that the following prerequisites are in place:
+
+- Shopify admin token is accessible.
+- The local folder containing the JSON file is accessible.
+
+# Implementation and Usage Notes
+
+This template reads the contents of a local JSON file and transforms the resulting JString to JToken. For every product that exists in the JSON file, the template will get the product information from Shopify and match it with the product title and stock keeping unit. If products are found in Shopify, the template will update the records with the contents of the local file. Otherwise the products will be added as new products to Shopify.
+Process variables include the full file path to the local JSON file, the base url for Shopify and the admin token for Shopify.
```
Comment (jefim) [inline Templates/Files/Local Files/JSON to Shopify - Products/long-description.md:16]: double newline

File: Templates/Files/Local Files/JSON to Shopify - Products/long-description.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+This template reads products from a local JSON file and imports them to Shopify.
+Using this template, you can import your local JSON product data to Shopify.
```
Comment (jefim) [inline Templates/Files/Local Files/JSON to Shopify - Products/long-description.md:2]: double newline

File: Templates/Files/Local Files/JSON to Shopify - Products/long-description.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            ]
+        }
+    ]
+}
+```
+
+# Error Handling
+
+The product handling tasks are followed by error checks. If an error occurs while handling a product, the handling for that product will stop and the next product will be taken for handling. Any exceptions encountered will be shown at the end of the process along with the counters for successful imports and updates.
+If transient errors are expected, retries for connections to Shopify can be enabled from the tasks. Transient errors are not handled.
```
Comment (jefim) [inline Templates/Files/Local Files/JSON to Shopify - Products/long-description.md:52]: double newline

File: Templates/Files/Local Files/JSON to Shopify - Products/metadata.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "JSON to Shopify - Products",
+    "version": "1.0.0",
+    "tags": [],
+    "additionalSystemsUsed": [
+    ]
```
Comment (jefim) [inline Templates/Files/Local Files/JSON to Shopify - Products/metadata.json:6]: mention Shopify, as we use plain HTTP calls for that

**Review by jefim — CHANGES_REQUESTED** [review]

**Review by jannejjj — APPROVED** [review]
The newlines have been added and the metadata adjusted.

**Pull request merged**